### PR TITLE
[#69] show sources on our server

### PR DESCRIPTION
**Pull request by Som3**

### CODE REVIEW

File: core/src/main/scala/codesearch/core/index/LanguageIndex.scala [hunk 1]
```diff
@@ -141,21 +153,16 @@ trait LanguageIndex[VTable <: DefaultTable] { self: DefaultDB[VTable] =>
...
-                          lineNumber - 1,
-                          rows
-                        )))
-      case _ =>
-        None
+    packageName(relativePath).map { p =>
+      val (firstLine, rows) = Helper.extractRows(relativePath, lineNumber)
+      CSearchResult(p,
+                    CodeSnippet(
+                      relativePath.split('/').drop(4).mkString("/"),
```
Comment (neongreen) [inline core/src/main/scala/codesearch/core/index/LanguageIndex.scala]: Can you show in comments how these paths look like? It's hard to follow `drop(4)`

File: web-server/app/views/search.scala.html [hunk 2]
```diff
@@ -1,18 +1,10 @@
-@import codesearch.core.index.LanguageIndex.PackageResult
-
-@(updated: String, packages: Seq[PackageResult], query: String, insensitive: Boolean, precise: Boolean, sources: Boolean, page: Int, totalMatches: Int, callURI: String)
+@(updated: String, lang: String)
 
 @headExtra = {
-    <meta http-equiv="Content-Security-Policy" content="default-src *; style-src 'self' 'unsafe-inline'; script-src 'self' 'unsafe-inline' 'unsafe-eval' http://www.google.com">
-
-    <script src="@routes.Assets.versioned("prismjs.js")"></script>
-    <link rel="stylesheet" type="text/css" href="@routes.Assets.versioned("css/prismjs.css")">
```
Comment (neongreen) [inline web-server/app/views/search.scala.html]: Why did prismjs go away?

Comment (Som3) [reply web-server/app/views/search.scala.html]: @neongreen `search` its page without code snippets (only search form)

File: core/src/main/scala/codesearch/core/index/LanguageIndex.scala (hunk 1 above)
Comment (Som3) [reply core/src/main/scala/codesearch/core/index/LanguageIndex.scala]: ok

File: web-server/app/codesearch/web/controllers/BaseController.scala [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+import com.github.marlonlom.utilities.timeago.TimeAgo
+import play.api.mvc.InjectedController
+import cats.instances.future._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+  * @author sss3 (Vladimir Alekseev)
+  */
+trait BaseController[V <: DefaultTable, I <: LanguageIndex[V]] { self: InjectedController =>
```
Comment (ppressives) [inline web-server/app/codesearch/web/controllers/BaseController.scala]: The trait name misleads cause of it's not base trait for all controllers, but only for search controllers

File: core/src/main/scala/codesearch/core/util/Helper.scala [hunk 4]
```diff
@@ -55,6 +57,13 @@ object Helper {
     }
   }
 
+  def extractFile(path: String)(implicit ec: ExecutionContext): Future[Option[List[String]]] =
+    Future {
+      Try {
+        Source.fromFile(path, "UTF-8").getLines
```
Comment (ppressives) [inline core/src/main/scala/codesearch/core/util/Helper.scala]: `scala.io.Source` uses blocking IO, so `extractFile` should be called with special ExecutionContext, if you want to use it with Future

File: core/src/main/scala/codesearch/core/util/Helper.scala [hunk 5]
```diff
@@ -55,6 +57,13 @@ object Helper {
     }
   }
 
+  def extractFile(path: String)(implicit ec: ExecutionContext): Future[Option[List[String]]] =
```
Comment (ppressives) [inline core/src/main/scala/codesearch/core/util/Helper.scala]: Hard to understand by function name that it just reads lines from file.

File: web-server/app/codesearch/web/controllers/BaseController.scala (hunk 3 above)
Comment (Som3) [reply web-server/app/codesearch/web/controllers/BaseController.scala]: U are right :+1:
